packages/workflow/src/Extensions/DateExtensions.ts

```diff
@@ -70,41 +71,34 @@ const DATETIMEUNIT_MAP: Record<string, DateTimeUnit> = {
 	ms: 'millisecond',
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function isDateTime(date: any): date is DateTime {
-	if (date) {
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-		return DateTime.isDateTime(date);
-	}
-	return false;
+function isLuxonDateTime(date: unknown): date is LuxonDateTime {
+	return date ? LuxonDateTime.isDateTime(date) : false;
 }
 
+const toLuxonDateTime = (d: Date | LuxonDateTime) =>
+	LuxonDateTime.isDateTime(d) ? d : LuxonDateTime.fromJSDate(d);
```

Comment:
What happens if `d` is just a string in ISO format? Would this also work?

Reply:
To my understanding that cannot happen because date extension methods are only callable on native JS date or Luxon datetimes. A string in ISO format would get string extension methods.

https://github.com/n8n-io/n8n/blob/39c871d5146e352258c41f5b425e317bb501df76/packages/workflow/src/Extensions/ExpressionExtension.ts#L470-L471

Reply:
Hum I think this is bad, as chaining won't work well.

See the example 
![Screenshot 2023-06-09 at 14 01 32](https://github.com/n8n-io/n8n/assets/219272/40c100d0-f919-4b38-9391-5041d184f6f4)

So the input is just a string that represents a date. I can't easily chain the functions. I have to call `.toDate()` before every chain step, and this is bad IMO. We should have a copy of all date functions implemented to strings as well or maybe think of a better approach, but I don't think I like the current solution. Maybe product can chip in with the intended behaviour here to clarify.

Reply:
Tracking [here](https://linear.app/n8n/issue/N8N-6252#comment-849917b4).

Reply:
Marking as resolved since we updated requirements!